src/test/java/com/google/sps/SyncServletTest.java

```diff
@@ -0,0 +1,119 @@
+        videos.add(Video.createVideo("This is a url"));
+        Room room = Room.createRoom(new LinkedList(),videos, new LinkedList());
+        long expected = newTimestamp;
+
+        room.updateCurrentVideoState(newState, newTimestamp);
+        long actual = room.getVideos().peek().getCurrentTimeStamp();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testVideoUpdateState(){
+        long newTimestamp = 50;
+        Video.VideoState newState = Video.VideoState.PLAYING;
+        Queue<Video> videos = new LinkedList<Video>();
+        videos.add(Video.createVideo("This is a url"));
+        Room room = Room.createRoom(new LinkedList(), videos, new LinkedList());
+        Video.VideoState expected = newState;
+
+        room.updateCurrentVideoState(newState, newTimestamp);
+        Video.VideoState actual = room.getVideos().peek().getCurrentState();
+        
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testVideoSyncWithVideoEnded(){
+        HttpServletRequest req = Mockito.mock(HttpServletRequest.class);
+        HttpServletResponse res = Mockito.mock(HttpServletResponse.class);
+        //Room that is being passed through the helper function
+        Room room = new Room(new LinkedList<Member>());
+        room.addVideo(Video.createVideo("www.Ross.com"));
+        room.addVideo(Video.createVideo("www.Johnson.com"));
+        //Room to have the expected properties when we test
+        Room room2 = new Room(new LinkedList<Member>());
+        room2.addVideo(Video.createVideo("www.Johnson.com"));
+        when(req.getParameter(UPDATE_STATE_PARAMETER)).thenReturn("0");
+
+        String expected = SyncServlet.roomToVideoJson(room2);
+        String actual = syncServletPost(req, res, room);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testVideoSyncWithVideoOngoing(){
+        HttpServletRequest req = Mockito.mock(HttpServletRequest.class);
+        HttpServletResponse res = Mockito.mock(HttpServletResponse.class);
+        //Room that is being passed through the helper function
+        Room room = new Room(new LinkedList<Member>());
+        room.addVideo(Video.createVideo("www.Ross.com"));
+        room.addVideo(Video.createVideo("www.Johnson.com"));
+        //Room to have the expected properties when we test
+        Room room2 = new Room(new LinkedList<Member>());
+        room2.addVideo(Video.createVideo("www.Ross.com"));
+        room2.updateCurrentVideoState(Video.VideoState.PAUSED,1234);
+        when(req.getParameter(UPDATE_STATE_PARAMETER)).thenReturn("2");
+        when(req.getParameter(VIDEO_TIMESTAMP_PARAMETER)).thenReturn("1234");
+
+        String expected = SyncServlet.roomToVideoJson(room2);
+        String actual = syncServletPost(req, res, room);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    private String syncServletPost(HttpServletRequest req, HttpServletResponse res, Room room){
```

Comment:
This seems strange. Why you are testing a function in the test class not the actual implementation logic?

Reply:
This is because I have to mock the room since it is being retrieved from datastore in the actual implementation.